cmd/describe.go

```diff
@@ -0,0 +1,70 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/redhat-developer/ocdev/pkg/application"
+	"github.com/redhat-developer/ocdev/pkg/component"
+	"github.com/redhat-developer/ocdev/pkg/occlient"
+	"github.com/redhat-developer/ocdev/pkg/project"
+	"github.com/redhat-developer/ocdev/pkg/storage"
+	"github.com/redhat-developer/ocdev/pkg/url"
+	"github.com/spf13/cobra"
+)
+
+var describeCmd = &cobra.Command{
+	Use:   "describe [component_name]",
+	Short: "Describe the given component",
+	Long:  `Describe the given component.`,
+	Example: ` # Describe nodejs component,
+	ocdev describe nodejs
+	Component nodejs with source in https://github.com/kadel/nodejs-ex
+	This Component is externally exposed via nodejs-myproject.192.168.42.231.nip.io
+	This Component uses storage which is nodestorage of size 1Gi
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client := getOcClient()
+
+		var cmpnt string
+		if len(args) == 0 {
+			var err error
+			cmpnt, err = component.GetCurrent(client)
+			checkError(err, "")
+		} else {
+			cmpnt = args[0]
+		}
+		GetComponentDesc(client, cmpnt)
+	},
+}
+
+// Get Component Description
+func GetComponentDesc(client *occlient.Client, cmpnt string) {
```

Comment:
https://github.com/redhat-developer/odo/pull/352 was merged, it should be possible to move this to `component` package